.github/workflows/test_pr_changes.yml

```diff
@@ -16,21 +16,15 @@ jobs:
   # We'll use the results to run tests only for changed frameworks.
   detect-changes:
     name: Detect PR Changes
-    if: >
-      github.event.action != 'edited' || (
-        github.event.changes.title == null &&
-        github.event.changes.body == null &&
-        github.event.changes.assignees == null &&
-        github.event.changes.reviewers == null &&
-        github.event.changes.labels == null
-      ) && github.event.pull_request.draft == false
+    if: ${{ github.event.pull_request.draft != true }}
     runs-on: ubuntu-24.04
     # Set job outputs to values from filter step
     outputs:
       folders: ${{ steps.filter.outputs.changes }}
 
     steps:
       - uses: actions/checkout@v4
+        if: ${{ github.event_name == 'merge_group' }}
```

Comment:
Checkout is only required for merge_group, for pull_request the API is used: https://github.com/dorny/paths-filter?tab=readme-ov-file#supported-workflows

Reply:
did you test this? the doc example shows otherwise or I don't get it

Reply:
At least for the SwiftLint and SwiftFormat workflows in this PR it seems to work, but they use v3 of this action.
<img width="1366" alt="Screenshot 2025-01-08 at 12 56 50" src="https://github.com/user-attachments/assets/587cf461-0e97-4f09-af10-86c891f3d9be" />